avm/ptn/azd/aks/main.bicep

```diff
@@ -0,0 +1,465 @@
+]
+
+// ============== //
+// Resources      //
+// ============== //
+
+#disable-next-line no-deployments-resources
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.azd-aks.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2021-12-01-preview' existing = if (!empty(logAnalyticsName)) {
+  name: logAnalyticsName
+}
+
+module managedCluster 'br/public:avm/res/container-service/managed-cluster:0.3.0' = {
+  name: '${uniqueString(deployment().name, location)}-managed-cluster'
+  params: {
+    name: name
+    location: location
+    tags: tags
+    nodeResourceGroup: nodeResourceGroupName
+    networkDataplane: networkDataplane
+    networkPlugin: networkPlugin
+    networkPluginMode: networkPluginMode
+    networkPolicy: networkPolicy
+    podCidr: podCidr
+    serviceCidr: serviceCidr
+    dnsServiceIP: dnsServiceIP
+    kubernetesVersion: kubernetesVersion
+    sshPublicKey: sshPublicKey
+    aadProfileEnableAzureRBAC: aadProfileEnableAzureRBAC
+    skuTier: skuTier
+    appGatewayResourceId: appGatewayResourceId
+    monitoringWorkspaceId: monitoringWorkspaceResourceId
+    managedIdentities: {
+      systemAssigned: true
+    }
+    diagnosticSettings: [
+      {
+        logCategoriesAndGroups: [
+          {
+            category: 'cluster-autoscaler'
+            enabled: true
+          }
+          {
+            category: 'kube-controller-manager'
+            enabled: true
+          }
+          {
+            category: 'kube-audit-admin'
+            enabled: true
+          }
+          {
+            category: 'guard'
+            enabled: true
+          }
+        ]
+        workspaceResourceId: !empty(logAnalyticsName) ? logAnalytics.id : ''
+        metricCategories: [
+          {
+            category: 'AllMetrics'
+            enabled: true
+          }
+        ]
+      }
+    ]
+    webApplicationRoutingEnabled: webApplicationRoutingEnabled
+    primaryAgentPoolProfile: primaryAgentPoolProfile
+    dnsPrefix: dnsPrefix
+    agentPools: agentPools
+    roleAssignments: [
+      {
+        name: aksClusterRoleAssignmentName
+        principalId: principalId
+        principalType: principalType
+        roleDefinitionIdOrName: aksClusterAdminRole
+      }
+    ]
+  }
+}
+
+module containerRegistry 'br/public:avm/res/container-registry/registry:0.4.0' = {
+  name: '${uniqueString(deployment().name, location)}-container-registry'
+  params: {
```

Comment:
Misses passthru of `enableTelemetry` parameter

Reply:
Not the correct fix. Please don't add the telemetry parameter as a condition, but just pass the parameter in the params block through to the linked module. Also please don't self-resolve comments 😉

Reply:
I have updated by your comments.